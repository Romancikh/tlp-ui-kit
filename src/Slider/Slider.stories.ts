import type { Meta, StoryObj } from "@storybook/react";

import Slider from "./Slider";

const meta = {
  component: Slider,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
  title: "Slider",
} satisfies Meta<typeof Slider>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    max: 5,
    min: 1,
    width: "250px",
  },
};
