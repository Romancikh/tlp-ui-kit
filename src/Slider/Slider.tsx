/* eslint-disable no-unused-vars */
import MuiSlider from "@mui/material/Slider";

type SliderProps = {
  marks?: boolean;
  max?: number;
  min?: number;
  name?: string;
  onChange?: (
    event: Event,
    value: number | number[],
    activeThumb: number,
  ) => void;
  value?: number;
  valueLabelDisplay?: "auto" | "off" | "on";
  width?: string;
};

function Slider({
  marks,
  max,
  min,
  name,
  onChange,
  value,
  valueLabelDisplay,
  width,
}: SliderProps) {
  return (
    <MuiSlider
      marks={marks}
      max={max}
      min={min}
      name={name}
      onChange={onChange}
      sx={{ width }}
      value={value}
      valueLabelDisplay={valueLabelDisplay}
    />
  );
}

export default Slider;
