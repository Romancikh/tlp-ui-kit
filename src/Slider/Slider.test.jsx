import { render } from "@testing-library/react";
import Slider from "./Slider";

test("Slider renders correctly", () => {
  const { container } = render(<Slider min={1} max={5} width="250px" />);
  expect(container).toMatchSnapshot();
});
