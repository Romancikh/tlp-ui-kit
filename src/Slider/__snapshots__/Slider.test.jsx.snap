// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider renders correctly 1`] = `
<div>
  <span
    class="MuiSlider-root MuiSlider-colorPrimary MuiSlider-sizeMedium css-zugq7j-MuiSlider-root"
  >
    <span
      class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
    />
    <span
      class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
      style="left: 0%; width: 0%;"
    />
    <span
      class="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary css-eg0mwd-MuiSlider-thumb"
      data-index="0"
      style="left: 0%;"
    >
      <input
        aria-orientation="horizontal"
        aria-valuemax="5"
        aria-valuemin="1"
        aria-valuenow="1"
        data-index="0"
        max="5"
        min="1"
        step="1"
        style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
        type="range"
        value="1"
      />
    </span>
  </span>
</div>
`;
